app/models/item.rb

```diff
@@ -1,3 +1,13 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def supermarket_name
+    supermarket.name
+  end
+
+  def customer_count
+    customers.count
```

Comment:
Excellent encapsulation of this calculation! I know it's a simple calculation, but according to strict MVC principles, it's beyond basic CRUD functionality and updates the SQL query so it's best to put it in a model method :)